tweet_recommendations/utils/metrics.py

```diff
@@ -19,5 +21,7 @@ def average_precision_at_k(actual, predicted, k=5):
     return score / min(len(actual), k)
 
 
-def mean_average_precision_at_k(actual, predicted, k=5):
-    return np.mean([average_precision_at_k(a, p, k) for a, p in zip(actual, predicted)])
+def mean_average_precision_at_k(actual: List[Any], predicted: List[Any], k=5) -> float:
+    return float(
+        np.mean([average_precision_at_k(a, p, k) for a, p in zip(actual, predicted)])
```

Comment:
Czyli jak wyglądają predykcje teraz? To jest lista list? Bo jeśli to jest płaska lista to liczysz AP@k dla pojedynczych elementów, co nie ma sensu.

Reply:
AP@K dotyczy pojedynczego przykładu -> expected = [1,3,5,7], predicted = [1, 5, 3, 7]
mAP@K dotyczy listy takich przykładów -> List[expected], List[predicted]
Nie wiem czy to cokolwiek wyjaśnia.